frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java

```diff
@@ -40,172 +23,39 @@ public class SlideOutBackgroundable extends Task.Backgroundable {
-
-    new Task.Backgroundable(project, getString("action.GitMachete.BaseSlideOutAction.task.title")) {
-      @Override
-      @UIThreadUnsafe
-      public void run(ProgressIndicator indicator) {
-        if (slideOutOptions != null) {
-          handleBranchDeletionDecision(branchName, gitRepository, slideOutOptions.shouldDelete());
-
-          if (slideOutOptions.shouldRemember()) {
-            val value = String.valueOf(slideOutOptions.shouldDelete());
-            setDeleteLocalBranchOnSlideOutGitConfigValue(gitRepository.getRoot(), value);
-          }
-        } else {
-          val title = getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-info.canceled");
-          val message = getString(
-              "action.GitMachete.BaseSlideOutAction.notification.message.slide-out-info.canceled.HTML")
-                  .fmt(branchName);
-          VcsNotifier.getInstance(project).notifyInfo(/* displayId */ null, title, message);
-        }
-      }
-    }.queue();
-  }
-
-  @UIThreadUnsafe
-  private void slideOutBranch(String branchName) {
-    val theSelectedGitRepository = this.selectedGitRepository;
-    if (branchLayout == null || theSelectedGitRepository == null) {
-      return;
-    }
-
-    LOG.info("Sliding out '${branchName}' branch in memory");
-    val newBranchLayout = branchLayout.slideOut(branchName);
-
-    runWriteActionOnUIThread(() -> {
-      try {
-        val macheteFilePath = getMacheteFilePath(theSelectedGitRepository);
-        LOG.info("Writing new branch layout into ${macheteFilePath}");
-        MacheteFileWriter.writeBranchLayout(macheteFilePath, branchLayoutWriter,
-            newBranchLayout, /* backupOldFile */ true, /* requestor */ this);
-
-      } catch (IOException e) {
-        val exceptionMessage = e.getMessage();
-        val errorMessage = "Error occurred while sliding out '${branchName}' branch" +
-            (exceptionMessage == null ? "" : ": " + exceptionMessage);
-        LOG.error(errorMessage);
-        VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-            getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-fail.HTML").fmt(branchName),
-            exceptionMessage == null ? "" : exceptionMessage);
-      }
-    });
-  }
-
-  @UIThreadUnsafe
-  private void handleBranchDeletionDecision(String branchName, GitRepository gitRepository, boolean shouldDelete) {
-    slideOutBranch(branchName);
-    if (shouldDelete) {
-      GitBrancher.getInstance(project).deleteBranch(branchName, Collections.singletonList(gitRepository));
-      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
-          /* title */ "",
-          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.with-delete.HTML")
-              .fmt(branchName));
-      return;
+    if (gitRepository != null) {
```

Comment:
I am thinking... does it even make sense to distinguish the case when gitRepository is null? the run does nothing it such case 🤔 

maybe let's make gitRepository not-null in this scope already, and bring the SlideOutAndDelete handler back here? wdyt @PawelLipski ?

Reply:
+1, I'd go for passing `gitRepository` to this class as `@NonNull` already